File: Python3/Medium/Accepted/average-waiting-time_6.py
class Solution:
    def averageWaitingTime(self, customers: List[List[int]]) -> float:
        
        t = 0
        s = 0
        for i in range(len(customers)):
            if t > customers[i][0]:
                t += customers[i][1]
                s += t - customers[i][0]
            else:
                t = customers[i][0] + customers[i][1]
                s += customers[i][1]
        return s/len(customers)

File: C++/Medium/Accepted/average-waiting-time_6.cpp
class Solution {
public:
    double averageWaitingTime(vector<vector<int>>& customers) {
        
        double res = 0;
        int n = customers.size();
        int curTime = 0;
        for (int i = 0; i < n; i++) {
            int arrival = customers[i][0];
            int time = customers[i][1];
            if (arrival > curTime) {
                curTime = arrival;
            }
            curTime += time;
            res += curTime - arrival;
        }
        return res / n;
    }
};